### ISSUE TITLE
landsat terralook has an unnatural hue

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#874]
I'm showing an example mentioned at https://github.com/dankelley/oce/issues/868#issuecomment-183874389; note how green everything is, even the water.  (See the link for data source.)

![screen shot 2016-02-14 at 6 55 27 am](https://cloud.githubusercontent.com/assets/99469/13033321/644c018c-d2e9-11e5-9bad-20bd539f75a5.png)

**Comment by dankelley** [comment]
@richardsc -- have you also found that `band="terralook"` gives odd colours? I came across this because I was thinking about #868 and wanted to take a peak at the "colour" channel.

PS. I am making `read.landsat()` permit `band=NULL` to just read the geometry info. That speeds things up hugely when all you need is to see where the image is from, generally. I may also make an issue that `read.landsat()` should take a latlon box, because reading all the band data takes a long time even on my fast home machine.

**Comment by richardsc** [comment]
I have noticed that often the `terralook` images come out quite green. I think tweaking band combinations to get a "true colour" image will vary from image to image, but perhaps there is better algorithm than the one we're using. I usually don't worry too much because I often an trying to get the `temperature` band. I suppose if we figure out pan-sharpening it would be good to have a decent "true colour" algorithm for high-res images.

**Comment by dankelley** [comment]
Isn't it wrong to try to pan-sharpen a temperature band? With the colour-type bands we can imagine that panchromatic is covering the same wavelengths, so if there is small-scale information in panchromatic, then it can be used to refine the R, G, and B bands. But thermal is something else, isn't it?  I'm not talking about the programming level, but rather the science level.  Although I am not arguing that pan-sharpening a thermal band is silly -- that was my intention when I started this issue -- I am just a bit worried about spending a lot of time coding something that might not be too meaningful.

As you can tell from the issue reports, I've been twiddling with the code all morning, partly trying to get an idea of how complex this would be, coding-wise.

**A favour** if you want to help out in this, the best would be to find alternative formulae for natural colour.  Also, I guess we should admit that the colours in the image shown in the first comment could well be pretty close to correct on this scale. The hue is slightly off but when you fly over NS, you mainly see greenish. And when you fly over water, you often don't see a whole lot of colour variation. So the "terralook" may be fine.  It may only be when you zoom on a concrete city that you start to see things.

**Comment by dankelley** [comment]
I made a test, zooming in on Halifax. The code and test .rda file are in the [issues site](https://github.com/dankelley/oce-issues/blob/master/868) and it produces as below, with a late-June image of a region that CR and I know very well.

Panel A is the present default. To my eye, it is too green. Look especially down by pier 21, and inshore of there (the train-track region). This region is covered by asphalt or gravel, and it should be gray-brown. Also, the water is arguably too green, although I don't know about this, really, given sun angle etc (and this *is* June).

In the other panels, I gradually reduce the `green.f` value from the default of 2. To my eye, values of 1.8 and 1.7 seem more natural.

I really cannot decide whether to change the default value. Probably a larger test suite is required, since I don't like changing defaults without a pretty good reason. (My preferred approach is to simply discuss the defaults in the documentation.)

I may blog about this, since I wouldn't mind it if other users weighed in.

Actually, possibly Clark could show this kind of thing in the workshop.  I think it's pretty neat how R gives us these tools for adjusting colour.

Below is the code as it exists at this moment (I *do* tend to twiddle things!) and the graph as of this moment.
```R
library(oce)
##> if (!length(list.files(pattern="^lt.rda$"))) {
##>     l <- read.landsat('/data/archive/landsat/LC80070292014170LGN00/')
##>     lt <- landsatTrim(l, ll=c(-63.60, 44.62), ur=c(-63.55, 44.655))
##>     save(lt, file="lt.rda")
##> } else {
##>     load("lt.rda")
##> }

load("lt.rda")

if (!interactive()) png("868a.png", width=7, height=7, unit="in", res=100)
par(mfrow=c(2,2), mar=c(2,2,2,2))
cex <- 1

plot(lt, band='terralook', mar=c(0.25,0.25,1,0.25), axes=FALSE)
mtext(lt[['time']], side=3, line=0, adj=1, cex=cex*0.9, col='blue')
mtext("A. default", side=3, line=0, adj=0, cex=cex)

plot(lt, band='terralook', green.f=1.9, mar=c(0.25,0.25,1,0.25), axes=FALSE)
mtext("B. green.f=1.9", side=3, line=0, adj=0, cex=cex)

plot(lt, band='terralook', green.f=1.8, mar=c(0.25,0.25,1,0.25), axes=FALSE)
mtext("C. green.f=1.8", side=3, line=0, adj=0, cex=cex)

plot(lt, band='terralook', green.f=1.7, mar=c(0.25,0.25,1,0.25), axes=FALSE)
mtext("D. green.f=1.7", side=3, line=0, adj=0, cex=cex)

if (!interactive()) dev.off()
```



![868a](https://cloud.githubusercontent.com/assets/99469/13186069/a7483a70-d71a-11e5-92c9-92e012f39d55.png)

**Comment by dankelley** [comment]
I've made some progress on this and will make up a blog posting to explain better what I've done (more images) and what I think are the best defaults. A posting is nice because it can be organized better than a GH issue. Most likely I'll post tomorrow or Sunday.

**Comment by dankelley** [comment]
I have written a blog posting on this, at http://dankelley.github.io/r/2016/02/20/landsat-hue.html and I invite comments here on that posting.  I realize that the two people who are likely to make those comments are going to be away for a while at AGU, so I plan to leave the issue open for a week or two, even though I am pretty sure I have come up with much better defaults (again, see the blog).

**Comment by dankelley** [comment]
Fixed in `develop` commit `cda829efa19dce4dc635f507816a905ae27d3a46` based on the http://dankelley.github.io/r/2016/02/21/landsat-hue-2.html blog posting.
